src/supervisor/policyAnalysis.py

```diff
@@ -0,0 +1,194 @@
+        root_policy_number_trend_df.plot(title="Count of Root Policies",figsize=(30, 15))
+        plt.savefig('../../output/policy-root-count.png')
+        saveCSV(root_policy_number_trend_df,"policy-root-count",True)
+        plt.close('all') 
+
+    except Exception as e:
+        print(Fore.RED+"Error in getting number of root policies: ",e)
+        print(Style.RESET_ALL)    
+    print("=============================================")
+   
+    status=True
+    return status
+
+def getReplicatedPolicy(pc,startTime, endTime, step):
+    print("Get details of the replicated policies")
+
+    try:
+        propagated_policy = pc.custom_query('count(policy_governance_info{type="propagated"})')
+
+        propagated_policy_df = MetricSnapshotDataFrame(propagated_policy)
+        propagated_policy_df["value"]=propagated_policy_df["value"].astype(float)
+        propagated_policy_df.rename(columns={"value": "NumOfPropagatedPolicies"}, inplace = True)
+        print(propagated_policy_df[['NumOfPropagatedPolicies']].to_markdown())
+
+        
+        propagated_policy_number_trend = pc.custom_query_range(
+        query='count(policy_governance_info{type="propagated"})',
+            start_time=startTime,
+            end_time=endTime,
+            step=step,
+        )
+
+        propagated_policy_number_trend_df = MetricRangeDataFrame(propagated_policy_number_trend)
+        propagated_policy_number_trend_df["value"]=propagated_policy_number_trend_df["value"].astype(float)
+        propagated_policy_number_trend_df.index= pandas.to_datetime(propagated_policy_number_trend_df.index, unit="s")
+        #node_cpu_trend_df =  node_cpu_trend_df.pivot( columns='node',values='value')
+        propagated_policy_number_trend_df.rename(columns={"value": "PropagatedtPolicyTotal"}, inplace = True)
+        propagated_policy_number_trend_df.plot(title="Total Count of propagated Policies",figsize=(30, 15))
+        plt.savefig('../../output/policy-propagated-count.png')
+        saveCSV(propagated_policy_number_trend_df,"policy-propagated-count",True)
+        plt.close('all') 
+
+    except Exception as e:
+        print(Fore.RED+"Error in getting number of propagated policies: ",e)
+        print(Style.RESET_ALL)    
+    print("=============================================")
+   
+    status=True
+    return status
+
+
+def getReplicatedPolicyControllerResponseTime(pc,startTime, endTime, step):
+    print("Get Percentage of replicated-policy reconcile times le 1 sec")
+
+    # what is the issue with
+    # histogram_quantile(0.95, sum(rate(controller_runtime_reconcile_time_seconds_bucket{controller="replicated-policy"}[30m])))
+    try:
+        propagated_policy = pc.custom_query('sum without(le,instance,pod) (controller_runtime_reconcile_time_seconds_bucket{controller="replicated-policy",le="1"})/sum without(le,instance,pod) (controller_runtime_reconcile_time_seconds_count{controller="replicated-policy"})')
+
```

Comment:
Till such time, here is next best metric that comes to my mind. @gparvin please weigh in:
It is the `number of reconciles done in less than equal to 1 sec / total number of reconciles`.
Did I get the controller name right?

Reply:
Yes, this is the correct controller that I think you want to focus on

Reply:
Thank You.